Update documentation about in-browser execution

choldgraf opened the pull request:
This refreshes the page about in-browser execution with Thebe. It does a few major things:

- Updates the documentation to be consistent in how we recommend configuration.
- Adds some examples of what the UI looks like to control execution, and what is possible.
- Streamlines the sections to follow the `how to` section title structure we've been using throughout.
- Generally reduce the amount of words and focus on action sentences and bullets.
- Activates in-browser computation for our documents so that we can

### Next steps

- [ ] Review the content again to make sure it is all correct
- [ ] Figure out why page-level over-rides aren't working. I tried this for just one page, but couldn't get it to work, which is why I switched to a project-level config.

stevejpurves reviewed (approved):
Looks great @choldgraf - see comments above

choldgraf commented on docs/in-browser-execution.md:
````diff
@@ -0,0 +1,351 @@
+```{attention} Be aware of multiple keys
+There are two possible locations for `binder` keys:
+
+1. `project.binder` is for [launch button configuration](./website-launch-buttons.md), and will take the user to a new page with a Binder session.
````
Sounds good - where's the issue where we can discuss the right config to use? Either way I'll leave this content as-is since it's the current reality.

choldgraf commented:
I've addressed the comment about JupyterHub - we should def improve that workflow so people can use their hubs to power their docs, but that can be a follow up enhancement. I've made one change, which is to change `in-browser-execution` to `in-page-execution`, since technically _everything_ in Jupyter interfaces is in-browser-execution and this disambiguates it a bit :-) 

I'll merge this one in, thanks all!
